controllers/postController.js

```diff
@@ -100,4 +102,159 @@ const list = async (req, res) => {
   }
 };
 
-module.exports = { upsert, list };
+const today = async (req, res) => {
+  if (!isValidString(req.params.keywordId) || isEmptyString(req.params.keywordId)) {
+    return res.status(400).send({ message: "[InvalidKeywordId] Error occured" });
+  }
+
+  const hasKeywordId = (await keywordModel.findOne({ _id: req.params.keywordId }).exec()) !== null;
+  if (!hasKeywordId) {
+    return res.status(400).send({ message: "[NotExistedKeywordId] Error occured" });
+  }
+
+  const keywordId = req.params.keywordId;
+
+  try {
+    const today = new Date();
+    const yesterday = new Date(today.getFullYear(), today.getMonth(), today.getDate() - 1);
+
+    const todayPostCount = await postModel
+      .find({ keywordId })
+      .find({
+        createdAt: {
+          $gte: today.setHours(0, 0, 0, 0),
+          $lt: today.setHours(23, 59, 59, 999),
```

Comment:
const로 선언된 Date 객체인 today를 `setHours()`를 통해 변경을 하고 있습니다.  today를 쓰기보다는 아래와 같이 new Date()를 써서 today가 변하지 않도록 하는 것이 좋을 것 같다고 생각했습니다.

```suggestion
          $gte: new Date().setHours(0, 0, 0, 0),
          $lt: new Date().setHours(23, 59, 59, 999),
```

Reply:
저도 규경님 의견에 동감해요~ 
`new Date()` 로 객체 `today`를 이미 만들긴 했지만 `setHours()` 메소드가 원본 `today`에 영향을 주기 때문이에요.

<img width="500" alt="Screenshot 2024-11-13 at 11 28 17 AM" src="https://github.com/user-attachments/assets/ebf21ccf-ab3c-443f-9786-51d77ad5d9ce">

Reply:
네 두 분 의견 반영해서 수정했습니다.